stock_picking_batch_extended/models/stock_picking.py

```diff
@@ -41,7 +41,16 @@ def action_done(self):
         """
         result = super(StockPicking, self).action_done()
         self.mapped('batch_id').verify_state()
-
+        picking_to_invoice_ids = self.env.context.get(
+            "picking_to_invoice_in_batch", [])
+        sales = self.filtered(
+            lambda r: r.id in picking_to_invoice_ids,
+        ).mapped("sale_id")
+        self.env['sale.advance.payment.inv'].create(
+            {'advance_payment_method': 'all'}
+        ).with_context(active_ids=sales.ids).create_invoices()
+        # for invoice in sales.invoice_ids:
+        sales.mapped("invoice_ids").action_invoice_open()
```

Comment:
this may fail, some sales already may have invoices in different states, paid, canceled, you need to filter them first